test/soldiers.test.js

```diff
@@ -0,0 +1,67 @@
+import {
+  afterAll,
+  beforeAll, beforeEach, describe, expect, it,
+} from 'vitest';
+import { client } from '../connections.js';
+import app from '../app.js';
+import { dbName, dbCollection, addNewSoldier } from '../database.js';
+
+beforeAll(async () => {
+  await client.connect();
+});
+
+afterAll(async () => {
+  await client.close();
+});
+
+beforeEach(async () => {
+  await client.db(dbName).collection(dbCollection).deleteMany({});
+});
+
+const testSoldier = {
+  _id: 3251084213556,
+  name: 'Shira',
+  rank: 3,
+  limitations: ['none', 'none'],
+};
+
+describe('Post soldier route', () => {
+  it('checks that status is 201', async () => {
+    const res = await app.inject({
+      method: 'POST',
+      url: '/soldiers',
+      body: testSoldier,
+    });
+    expect(res.statusCode).toBe(201);
+    expect(res.body).toEqual('soldier created');
+  });
+});
+
+describe('Get soldier by id route', () => {
+  it('checks that when soldier is found status is 200', async () => {
+    await addNewSoldier(client, testSoldier);
+    const res = await app.inject({
+      method: 'GET',
+      url: '/soldiers/3251084213556',
+    });
+    expect(res.statusCode).toBe(200);
+  });
+  it('checks that when soldier is not found status is 400', async () => {
+    const res = await app.inject({
+      method: 'GET',
+      url: '/soldiers/66666666',
+    });
+    expect(res.statusCode).toBe(400);
+  });
+});
+
+describe('Get soldiers route', () => {
```

Comment:
I think you need to add more test, because there is more logic to it

Reply:
added